sources/ecs-gpu-init/cmd/ecs-gpu-init/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+const (
+	// NvidiaGPUInfoFilePath is the file path where GPUs and driver info are saved
+	NvidiaGPUInfoFilePath = "/var/lib/ecs/gpu/nvidia-gpu-info.json"
+)
+
+func main() {
+	if err := generateGPUInfoConfig(); err != nil {
+		fmt.Fprintf(os.Stderr, "Got error %v", err)
+		os.Exit(1)
+	}
+}
+
+// GenerateGPUInfoConfig generates the configuration used by the ECS agent
+func generateGPUInfoConfig() error {
+	if ret := nvml.Init(); ret != nvml.SUCCESS {
+		return fmt.Errorf("Failed to initialize NVML, got ret %v", ret)
```

Comment:
nit: usual approach would be something like:
```suggestion
		return fmt.Errorf("Failed to initialize NVML: %v", ret)
```

(to avoid the "got ret" phrase which is superfluous and unconventional)